pkg/block/metadata/deletionmark.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package metadata
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"path"
+
+	"github.com/go-kit/kit/log"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+const (
+	// DeletionMarkFilename is the known json filename to store details about when block is marked for deletion.
+	DeletionMarkFilename = "deletion-mark.json"
```

Comment:
Technically this deletion mark belongs to block not metadata :thinking: Can we move it?

Reply:
Before resolving, worth to comment @khyatisoneji why this comment was resolved. Otherwise other contributors/maintainers can be confused why you "ignore" comments (:

Reply:
We are considering `deletion-mark.json` file as block metadata, so keeping this file in metadata package and not block package. Resolving this because of that.